### PR TITLE
Implement biz lic app replacement service

### PR DESCRIPTION
**Pull request by ruaraya**
Implementation of business licence application replacement service

### CODE REVIEW

File: src/Spd.Manager.Licence/BizLicAppManager.cs [hunk 1]
```diff
@@ -115,7 +115,44 @@ public async Task<BizLicAppCommandResponse> Handle(BizLicAppSubmitCommand cmd, C
...
+            new LicenceQry() { LicenceId = request.OriginalLicenceId },
+            cancellationToken);
+        if (originalLicences == null || !originalLicences.Items.Any())
+            throw new ArgumentException("cannot find the licence that needs to be replaced.");
+
+        BizLicApplicationResp originalLic = await _bizLicApplicationRepository.GetBizLicApplicationAsync((Guid)cmd.LicenceRequest.OriginalApplicationId, cancellationToken);
+        if (originalLic.BizId == null)
+            throw new ArgumentException("there is no business related to the application.");
+
+        var existingFiles = await GetExistingFileInfo(
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/BizLicAppManager.cs:133]: there should be no exsiting files. no need for code from 133 to 139

File: src/Spd.Manager.Licence/BizLicAppManager.cs [hunk 2]
```diff
@@ -115,7 +115,44 @@ public async Task<BizLicAppCommandResponse> Handle(BizLicAppSubmitCommand cmd, C
...
+                cmd.LicenceRequest.OriginalApplicationId,
+                cmd.LicenceRequest.PreviousDocumentIds,
+                cancellationToken);
+        await ValidateFilesForRenewUpdateAppAsync(cmd.LicenceRequest,
+            cmd.LicAppFileInfos.ToList(),
+            cancellationToken);
+
+        // Create new app
+        CreateBizLicApplicationCmd createApp = _mapper.Map<CreateBizLicApplicationCmd>(request);
+        createApp.UploadedDocumentEnums = GetUploadedDocumentEnums(cmd.LicAppFileInfos, existingFiles);
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/BizLicAppManager.cs:143]: no need for this

File: src/Spd.Manager.Licence/BizLicAppManager.cs [hunk 3]
```diff
@@ -115,7 +115,44 @@ public async Task<BizLicAppCommandResponse> Handle(BizLicAppSubmitCommand cmd, C
...
+        BizLicApplicationCmdResp response = await _bizLicApplicationRepository.CreateBizLicApplicationAsync(createApp, cancellationToken);
+
+        decimal cost = await CommitApplicationAsync(request, response.LicenceAppId, cancellationToken);
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/BizLicAppManager.cs:146]: commit should be done after updateMember

File: src/Spd.Manager.Licence/BizLicAppManager.cs (hunk 2 above)
Comment (ruaraya) [reply src/Spd.Manager.Licence/BizLicAppManager.cs]: Fixed.

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**